examples/audio/stft.py

```diff
@@ -0,0 +1,433 @@
+2. Apply `Conv1D` layers with `LayerNormalization` simialar to the
+   classic VGG design.
+4. Apply global maximum pooling to have fixed set of features.
+5. Add `Dense` layers to make the final predictions based on the features.
+"""
+
+model1d = keras.Sequential(
+    [
+        layers.InputLayer((None, 1)),
+        layers.STFTSpectrogram(
+            mode="log",
+            frame_length=SAMPLE_RATE * 40 // 1000,
+            frame_step=SAMPLE_RATE * 15 // 1000,
+            trainable=False,
+        ),
+        layers.Conv1D(64, 64, activation="relu"),
+        layers.Conv1D(128, 16, activation="relu"),
+        layers.LayerNormalization(),
+        layers.MaxPooling1D(4),
+        layers.Conv1D(128, 8, activation="relu"),
+        layers.Conv1D(256, 8, activation="relu"),
+        layers.Conv1D(512, 4, activation="relu"),
+        layers.LayerNormalization(),
+        layers.Dropout(0.5),
+        layers.GlobalMaxPooling1D(),
+        layers.Dense(256, activation="relu"),
+        layers.Dense(256, activation="relu"),
+        layers.Dropout(0.5),
+        layers.Dense(NUM_CLASSES, activation="softmax"),
+    ],
+    name="model_1d_non_trainble_stft",
+)
+model1d.compile(
+    optimizer=keras.optimizers.Adam(1e-5),
+    loss="sparse_categorical_crossentropy",
+    metrics=["accuracy"],
+)
+model1d.summary()
+
+"""
+Create the datasets.
+"""
+
+train_ds = read_dataset(pd_data, [1, 2, 3], BATCH_SIZE)
+valid_ds = read_dataset(pd_data, [4], BATCH_SIZE)
+
+"""
+Train the model and restore the best weights.
+"""
+
+history_model1d = model1d.fit(
+    train_ds,
+    validation_data=valid_ds,
+    epochs=EPOCHS,
+    callbacks=[
+        keras.callbacks.EarlyStopping(
+            monitor="val_loss",
+            patience=EPOCHS,
+            restore_best_weights=True,
+        )
+    ],
+)
+
+"""
+### Create the 2D model
+
+1. Create three spectrograms with multiple band-widths from the raw input.
+2. Concatenate the three spectrograms to have three channels.
+3. Load `MobileNet` and set the weights from the weights trained on `ImageNet`.
+4. Apply global maximum pooling to have fixed set of features.
+5. Add `Dense` layers to make the final predictions based on the features.
+"""
+
+input = layers.Input((None, 1))
+spectrograms = [
+    layers.STFTSpectrogram(
+        mode="log",
+        frame_length=SAMPLE_RATE * x // 1000,
+        frame_step=SAMPLE_RATE * 15 // 1000,
+        fft_length=2048,
+        padding="same",
+        expand_dims=True,
+        # trainable=True,  # trainable by default
+    )(input)
+    for x in [30, 40, 50]
+]
+
+multi_spectrograms = layers.Concatenate(axis=-1)(spectrograms)
+
+img_model_weights_no_top = keras.applications.MobileNet(
+    weights="imagenet"
+).get_weights()[:-2]
+
+img_model = keras.applications.MobileNet(
+    weights=None,
+    include_top=False,
+    pooling="max",
+)
+output = img_model(multi_spectrograms)
+img_model.set_weights(img_model_weights_no_top)
```

Comment:
Can you explain what you are doing in the past few lines? You should be able to instantiate MobileNet with pretrained weights without the top, in one line.

You could use a KerasHub backbone for this.